### PR TITLE
JVM Setup

### PR DESCRIPTION
**Pull request by buenaflor**
Sets up the project with commonMain / jvmMain. KFlogger is usable in this PR through commonMain on JVM apps.

Basic usage:

```kotlin
// in commonMain
import com.buenaflor.kflogger.KFluentLogger

fun log() {
  val logger = KFluentLogger.forEnclosingClass()
  logger.atWarning().log("log")
}
```

## Important info:
 - Java files (including tests) from Flogger were not modified expect for the package names in the import
 - Some java files such as `AbstractLogger`, `LoggingApi`, `LogSite` etc... have been ported to Kotlin JVM  for easier implementation of `expect/actual`
 - What did I modify / add:
   - Everything except files inside `jvmMain/java/com/buenaflor/kflogger` and `jvmTest/java/com/buenaflor/kflogger`
   
## 3rd part deps 
- errorprone plugin -> used within Flogger as well but the gradle plugin is an external contribution to errorprone but it is referenced on the google erroprone site [here](https://errorprone.info/docs/installation)
- spotless plugin -> The spotless plugin is used for easy usage of `ktfmt` formatting (personal preference, not used in Flogger). if necessary I can also remove/replace spotless.

### CODE REVIEW

**Comment by buenaflor** [comment]
@MarkCMann @dimart tagging you here cause I somehow can't add anyone as reviewer

File: api/src/jvmMain/kotlin/com/buenaflor/kflogger/LogSite.jvm.kt [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+ * Log sites are intended to be injected into code automatically, typically via some form of
+ * bytecode rewriting. Each injection mechanism can have its own implementation of `LogSite` adapted
+ * to its needs.
+ *
+ * As a fallback, for cases where no injection mechanism is configured, a log site based upon stack
+ * trace analysis is used. However due to limitations in the information available from
+ * `StackTraceElement`, this log site will not be unique if multiple log statements are on the the
+ * same, or if line number information was stripped from the class file.
+ */
+public actual abstract class LogSite : LogSiteKey {
```
Comment (MarkCMann) [inline api/src/jvmMain/kotlin/com/buenaflor/kflogger/LogSite.jvm.kt]: Are you able to type-alias these cases to their java equivalents?

File: api/src/jvmMain/kotlin/com/buenaflor/kflogger/AbstractLogger.jvm.kt [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+import java.util.concurrent.TimeUnit
+import java.util.logging.Level
+
+/**
+ * Base class for the fluent logger API. This class is a factory for instances of a fluent logging
+ * API, used to build log statements via method chaining.
+ *
+ * @param <API> the logging API provided by this logger. </API>
+ */
+public actual abstract class AbstractLogger<API : LoggingApi<API>>
```
Comment (MarkCMann) [inline api/src/jvmMain/kotlin/com/buenaflor/kflogger/AbstractLogger.jvm.kt]: Is this able to just be a typealias to the Java class?

File: api/src/jvmMain/kotlin/com/buenaflor/kflogger/LoggingApi.jvm.kt [hunk 3]
```diff
@@ -0,0 +1,1213 @@
...
+ * A hypothetical example of a context specific grammar might be:
+ * <pre>`logger.at(WARNING).whenSystem().isLowOnMemory().log("");
+ * `</pre> *
+ *
+ * In this example the `whenSystem()` method would return its own API with several context specific
+ * methods (`isLowOnMemory()`, `isThrashing()` etc...), however each of these sub-APIs must
+ * eventually return the original logging API.
+ */
+// NOTE: new methods to this interface should be coordinated with google-java-format
+public actual interface LoggingApi<API : LoggingApi<API>?> {
```
Comment (MarkCMann) [inline api/src/jvmMain/kotlin/com/buenaflor/kflogger/LoggingApi.jvm.kt]: This should be typealiased to the existing flogger version, I'm concerned that redefining this means that existing LoggingApi's will need to be rewritten (or at least modified) to say that they inherit from this new interface.

**Review by MarkCMann — COMMENTED** [review]
I'm not sure that we'll be able to easily adopt these changes without significant modifications in existing use cases of Flogger.

**Pull request merged**